### DISCUSSION TITLE
API query with filter in multiple nested table

### DISCUSSION THREAD

**Discussion by kapitani** [supabase/supabase#16461] [Questions]
Hi. I am new to Supabase but I am learning very hard everyday. I have a question regarding querying using filter for multiple nested table.

I have 3 tables.
1. material with fields = id, name, material_brand (relation to material_brand table)
2. material_brand with fields = id, brand_name, material_type (relation to material_type table)
3. material_type with fields = id, type_name

So far I have managed to query material table that filters material_brand (GET /material?select=*&material_brand=eq.[id]

But what I am trying to do is to query and filter based on material_type. How can I achieve this?

Appreciate all the helps, thank you.

**Comment by GaryAustin1** [comment]
https://postgrest.org/en/stable/references/api/resource_embedding.html?embedded-filters#embedded-filters
Goes thru many of the options to join tables and filter on them.

**Reply by kapitani** [reply]
Hi Gary, thanks alot for replying. I have read and tried many ways but to no avail. It seems like to reference does not have specifically like my tables, makes it harder to low-code developer like me. Attach is my tables. Can you help provide the example on how to GET material table and filter based on material_type's id field?

<img width="582" alt="Screenshot 2023-08-12 at 11 44 39 AM" src="https://github.com/supabase/supabase/assets/135631820/e473d43a-6f29-4fd3-869c-4d55707b9836">

**Reply by GaryAustin1** [reply]
I don't deal with the REST API directly myself...  This I would think be close: (not at all tested)
`/material?select=*,material_brand!inner(*,material_type!inner(*))&material_brand.material_type.type_name=eq.typename`

The !inner is used to filter the parent table also, which I assume was your goal.
